utils_test.go

```diff
@@ -68,7 +68,7 @@ func testCanCacheQuery(t *testing.T, q string, expected bool) {
 }
 
 func TestGetQuerySnippetGET(t *testing.T) {
-	req, err := http.NewRequest("GET", "", nil)
+	req, err := http.NewRequest("GET", "", bytes.NewBuffer(nil))
```

Comment:
this change shows the issue when `getQuerySnippet` panics if body is nil. Body could be nil for GET requests and proxy will panic in runtime:
```
	// For client requests, a nil body means the request has no
	// body, such as a GET request. 
```

Reply:
It's true only for the client mode but the server mode is not
https://golang.org/src/net/http/request.go?s=3211:11861#L174

Reply:
True, didn't know about that. However, I still find it useful to check object for not being nil before calling its methods. I think, it makes code more robust. 
Otherwise, you'll need to explain in this test why it is required to path non-nil body for GET request.

Reply:
Sure, I added check for request body isn't nil and fixed the test